internal/controllers/symlink-resolver/resolver.go

```diff
@@ -0,0 +1,43 @@
+package symlinkResolver
+
+import (
+	"path/filepath"
+	"sync"
+)
+
+type Resolver struct {
+	resolveFn resolverFunc
+	cache     sync.Map
+}
+
+type resolverFunc func(string) (string, error)
+
+var defaultResolverFn = filepath.EvalSymlinks
+
+func NewWithDefaultResolver() *Resolver {
+	return NewWithResolver(defaultResolverFn)
+}
+
+func NewWithResolver(resolveFn resolverFunc) *Resolver {
+	return &Resolver{
+		resolveFn: resolveFn,
+		cache:     sync.Map{},
+	}
+}
+
+func (r *Resolver) Resolve(path string) (string, error) {
+	var err error
+	val, ok := r.cache.Load(path)
+	if !ok {
+		val, err = r.resolveFn(path)
+		if err != nil {
+			return "", err
+		}
+		r.cache.Store(path, val)
+	}
+	return val.(string), nil
+}
+
+func (r *Resolver) Cleanup() {
```

Comment:
I dont think a resolver should be reusable and we should initiate it once per reconcile. If we think of resolver as Closeable, then this would look like:

```suggestion
func (r *Resolver) Close() error {
   r.cache = nil
   return nil
}
```

For GC thats a no-op so we might just completely drop this and initialize it from scratch in every reconcile from the resolveFn


Im also not sure if we need a sync.Map{} here, we are not in performance critical code, but this is more of a nit, so feel free to ignore.

Reply:
sounds okay to me even though it feels like premature optimization. resolving